platforms/core-execution/build-cache/src/integTest/groovy/org/gradle/caching/internal/ConfigurationCacheBuildCacheIntegrationTest.groovy

```diff
@@ -55,7 +55,7 @@ class ConfigurationCacheBuildCacheIntegrationTest extends AbstractIntegrationSpe
         fails "compileJava"
 
         then:
-        failureDescriptionStartsWith("Execution failed for task ':compileJava'.")
+        failureDescriptionStartsWith("Execution failed for task ':compileJava' (created by plugin class 'org.gradle.api.plugins.JavaBasePlugin').")
```

Comment:
❓ Is this because the `java-base` plugin is applied by class instead of by ID?

It is not a huge concern in general to display "by plugin class" information, but it seems to me that `compileJava` will be shown a lot to users and it might be a good opportunity to apply it by ID to make this message better.